transaction/core/src/validation/validate.rs

```diff
@@ -72,16 +72,34 @@ pub fn validate<R: RngCore + CryptoRng>(
     // Note: The transaction must not contain a Key Image that has previously been
     // spent. This must be checked outside the enclave.
 
+    // Each tx_out must conform to the structural rules for TxOut's at this block
+    // version
+    for tx_out in tx.prefix.outputs.iter() {
```

Comment:
Why do we only need to validate the txouts in the output and not the input?

Reply:
My take on it: The inputs are already in the ledger, and we don't know which block version they are a part of. Validation rules can change over time, but once something is in the ledger it should be spendable and is considered valid.

Reply:
lets answer this offline, this isn't directly relevant to this PR

the short answer is that there are no structural rules for inputs right now, they are validated when the signature is checked